transactor/src/main/scala/io/mediachain/datastore/DynamoDatastore.scala

```diff
@@ -0,0 +1,158 @@
+package io.mediachain.datastore
+
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
+import com.amazonaws.services.dynamodbv2.model.AttributeValue
+import java.nio.ByteBuffer
+import scala.collection.mutable.{Buffer, ArrayBuffer}
+import io.mediachain.multihash.MultiHash
+import io.mediachain.types.Datastore.DatastoreException
+
+// TODO error handling
+//      queued eventual writes in the background with disk backing of 
+//       in-progress writes
+//      use batch writes for chunked puts
+class DynamoDatastore(table: String, creds: BasicAWSCredentials)
+  extends BinaryDatastore with AutoCloseable {
+  import scala.collection.JavaConversions._
+  
+  val chunkSize = 1024 * 384 // 384 KB; DynamoDB has 400KB limit
```

Comment:
I really like https://github.com/twitter/util#space though I am guessing you will scoff at it 😛